sway-lib-std/src/i8.sw

```diff
@@ -0,0 +1,141 @@
+library i8;
+
+use core::num::*;
+use ::assert::assert;
+
+/// The 8-bit signed integer type.
+/// Represented as an underlying u8 value.
+pub struct i8 {
+    underlying: u8,
+}
+
+pub trait From {
+    /// Function for creating i8 from its u8 and bool components.
+    fn from(value: u8) -> Self;
+}
+
+impl From for i8 {
+    fn from(value: u8) -> i8 {
+        i8 {
+            underlying: value,
+        }
+    }
+}
+
+impl core::ops::Eq for i8 {
+    pub fn eq(self, other: i8) -> bool {
+        self.underlying == other.underlying
+    }
+}
+
+impl core::ops::Ord for i8 {
+    pub fn gt(self, other: Self) -> bool {
+        self.underlying > other.underlying
+    }
+
+    pub fn lt(self, other: Self) -> bool {
+        self.underlying < other.underlying
+    }
+}
+
+impl i8 {
+    pub fn indent() -> u8 {
+        128u8
+    }
+}
+
+impl i8 {
+    /// Initializes a new, zeroed i8.
+    pub fn new() -> i8 {
+        i8 {
+            underlying: ~i8::indent(),
+        }
+    }
+
+    /// The smallest value that can be represented by this integer type.
+    pub fn min() -> i8 {
+        i8 {
+            underlying: ~u8::min(),
+        }
+    }
+
+    /// The largest value that can be represented by this type,
+    pub fn max() -> i8 {
+        i8 {
+            underlying: ~u8::max(),
+        }
+    }
+
+    /// The size of this type in bits.
+    pub fn bits() -> u32 {
+        8
+    }
+
+    pub fn neg_from(value: u8) -> i8 {
+        i8 {
+            underlying: ~i8::indent() - value,
+        }
+    }
+
+    fn from_uint(value: u8) -> i8 {
+        let underlying: u8 = value + ~i8::indent(); // as the minimal value of i8 is -~i8::indent() (1 << 7) we should add ~i8::indent() (1 << 7) 
+        i8 {
+            underlying
+        }
+    }
+}
+
+impl core::ops::Add for i8 {
+    /// Add a i8 to a i8. Panics on overflow.
+    pub fn add(self, other: Self) -> Self {
+        ~i8::from(self.underlying - ~i8::indent() + other.underlying) // subtract 1 << 7 to avoid double move
```

Comment:
These comments are hard to read on smaller screens. Can you place them on top of instead of beside lines?